templates/vue/src/components/tyde/TydeMenuHome.vue

```diff
@@ -38,29 +42,60 @@ export default {
   data() {
     return {
       activeTab: "all",
+      logsWithAnswers: [],
       tabs: ["all", "activities", "content" /*, "favourites"*/],
     }
   },
   computed: {
     visibleLogs() {
       const filter = this.activeTab
       if (filter === "activities") {
-        return this.logs.filter(item => item.type === "activity")
+        return this.logsWithAnswers.filter(item => item.type === "activity")
       }
       if (filter === "content") {
-        return this.logs.filter(item => item.type === "content")
+        return this.logsWithAnswers.filter(item => item.type === "content")
       }
       if (filter === "favourites") {
-        return this.logs.filter(item => item.isFavourite)
+        return this.logsWithAnswers.filter(item => item.isFavourite)
       }
-      return this.logs
+      return this.logsWithAnswers
     },
   },
   methods: {
     setActiveTab(tab) {
       this.activeTab = tab
     },
+    async getAudioSrc(nodeId, audioId) {
+      try {
+        const audio = await TapestryApiClient.getH5PAudioFromServer(nodeId, audioId)
+        const blob = await Helpers.base64ToBlob(audio, 'audio/wav')
+        return URL.createObjectURL(blob)
+      } catch (e) {
+        console.error(e)
+        return ""
+      }
+    }
   },
+  watch: {
+    /**
+     * Further filter the logs to check if there are valid answers recorded
+     * on the server or not. If not, don't show those logs.
+     */
+    logs: {
+      immediate: true,
+      handler(newLogs) {
+        newLogs.forEach(async log => {
+          if (log.audioId) {
+            const audioSrc = await this.getAudioSrc(log.nodeId, log.audioId)
+            if (audioSrc)
+              this.logsWithAnswers.push({ ...log, audioSrc })
+          } else {
+            this.logsWithAnswers.push({ ...log })
+          }
```

Comment:
Did you test if this works correctly? From my understanding `async/await` does not work intuitively with `forEach`. The `forEach` will terminate without actually waiting for any of the promises to complete.

Reply:
Yes, it's correct that `forEach` will not wait for its items to be finished before moving to the next task. However, `forEach` is the only task here in this handler. Am I missing anything? And yes, I believe this is working.

Reply:
As long as nothing relies on these await calls and everything works as it should, then it's ok.